test/unit/test_conf.ml

```diff
@@ -0,0 +1,41 @@
+let test_check_version =
+  let test name ~expected ~exe res =
+    let test_name = "compare_version: " ^ name in
+    ( test_name
+    , `Quick
+    , fun () ->
+        let got = Ocamlformat_lib.Conf.check_version ~expected ~exe in
+        Alcotest.check Alcotest.(result unit string) test_name res got )
+  in
+  [ test "empty 'expected'" ~expected:"" ~exe:""
+      (Error "malformed version number \"\".")
+  ; test "invalid 'expected'" ~expected:"0.33..foo" ~exe:""
+      (Error "malformed version number \"0.33..foo\".")
+  ; test "invalid 'exe'" ~expected:"0.14.2" ~exe:"0.33..foo"
+      (Error
+         "expected ocamlformat version to be \"0.14.2\" but got \
+          \"0.33..foo\".")
+  ; test "unknown 'exe'" ~expected:"0.14.2" ~exe:"unknown"
+      (Error
+         "expected ocamlformat version to be \"0.14.2\" but got \"unknown\".")
+  ; test "equal" ~expected:"0.14.1" ~exe:"0.14.1" (Ok ())
+  ; test "accepted" ~expected:"0.14.1" ~exe:"0.14.1-15-g273f6f6-dirty"
+      (Ok ())
+  ; test "too old" ~expected:"0.14.1" ~exe:"0.14.0"
+      (Error
+         "expected ocamlformat version to be \"0.14.1\" but got \"0.14.0\". \
+          Please upgrade.")
+  ; test "too old 2" ~expected:"0.14.1" ~exe:"0.14.0-15-g273f6f6-dirty"
+      (Error
+         "expected ocamlformat version to be \"0.14.1\" but got \
+          \"0.14.0-15-g273f6f6-dirty\". Please upgrade.")
+  ; test "too recent" ~expected:"0.14.0" ~exe:"0.14.1"
+      (Error
+         "expected ocamlformat version to be \"0.14.0\" but got \"0.14.1\". \
+          Please downgrade.")
+  ; test "too recent 2" ~expected:"0.14.0" ~exe:"0.14.1-15-g273f6f6-dirty"
+      (Error
+         "expected ocamlformat version to be \"0.14.0\" but got \
+          \"0.14.1-15-g273f6f6-dirty\". Please downgrade.") ]
```

Comment:
when version number subsitution does not work for some reason (say, `git` is not installed), `gen_version.mlt` will put `"unknown"` in the executable. What do you think about adding a test for this case? And maybe having a special error message in that case as well?